File: array/python/remove_duplicates_from_sorted_array_ii.py
class Solution:
    def removeDuplicates2(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 1:
            return 1
        k = 2
        for i in range(2, n):
            if not ((nums[k - 2] == nums[k - 1]) and (nums[i] == nums[k - 1])):
                nums[k] = nums[i]
                k += 1
        return k

File: array/ts/remove_duplicates_from_sorted_array_ii.ts
function removeDuplicates2(nums: number[]): number {
    const n = nums.length;
    if (n === 1) return 1;
    let k = 2;
    for (let i = 2; i < n; i++) {
        if (!((nums[k - 2] === nums[k - 1]) && (nums[i] === nums[k - 1]))) {
            nums[k] = nums[i];
            k++;
        }
    }
    return k;
}
